config/routes.rb

```diff
@@ -1,12 +1,18 @@
 Rails.application.routes.draw do
-  get 'welcome/index'
+
+  # Basic authentication
+  get '/login',     to: 'sessions#new'
+  post '/login',    to: 'sessions#create'
+  delete '/logout', to: 'sessions#destroy'
 
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
   root 'welcome#index'
 
+  resources :users
```

Comment:
We implement only `index`, `show`,`new` and `create` method so it'd be great to restrict routes only to these actions.